### PR TITLE
read invite from url & check

### PR DESCRIPTION
**Pull request by taehyoungjo**

### CODE REVIEW

File: frontend/src/contexts/UserContext.tsx [hunk 1]
```diff
@@ -84,7 +106,12 @@ export const UserProvider: React.FC<{ children: React.ReactNode }> = ({
         [user, refreshUser, setCurrentWorkspace, setWorkspaces]
     )
 
-    return <UserContext.Provider value={store}>{children}</UserContext.Provider>
+    return (
+        <UserContext.Provider value={store}>
+            {children}
+            <Toaster />
```
Comment (kevinhu) [inline frontend/src/contexts/UserContext.tsx:112]: Why is the toaster specific to the user context? (as opposed to the app in general)

File: server/routers/workspaces.py [hunk 2]
```diff
@@ -121,6 +121,31 @@ async def list_user_workspaces(user: PrismaModels.User = Depends(get_user)):
     return workspaces
 
 
+@router.get(
+    "/workspace/list_users",
```
Comment (kevinhu) [inline server/routers/workspaces.py:125]: Just `/list_users` now—I added the prefix in the router

File: server/routers/users.py [hunk 3]
```diff
@@ -46,6 +48,18 @@ async def refresh_token(
     return result
 
 
+async def verify_token_with_auth0(
+    token: Any = Depends(token_auth_scheme),
+):
+    """ """
+    domain = os.getenv("AUTH0_DOMAIN", "https://your.domain.com/")
```
Comment (kevinhu) [inline server/routers/users.py:55]: Is the domain already defined in `config.py`?

File: server/routers/invites.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    role: Role,
+    user: PrismaModels.User = Depends(get_user),
+):
+    """
+    Create an invite for a user
+    """
+
+    await check_workspace_access(user.id, workspace_id)
+
+    salt = bcrypt.gensalt()
```
Comment (kevinhu) [inline server/routers/invites.py:29]: Let's use `passlib.context`, which is more secure here (see schedules.py for an example)

File: frontend/src/pages/Dashboard/Users.tsx [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    ADMIN = 'ADMIN',
+    USER = 'USER',
+}
+
+const roles = [
+    { value: UserRoles.ADMIN, label: 'Admin' },
+    { value: UserRoles.USER, label: 'User' },
+]
+
+const redirectUri = 'http://localhost:5173/app'
```
Comment (kevinhu) [inline frontend/src/pages/Dashboard/Users.tsx:95]: Should this be an env variable

File: server/routers/users.py (hunk 3 above)
Comment (taehyoungjo) [reply server/routers/users.py:55]: its in VerifyToken, not sure how to pull them over

Comment (kevinhu) [reply server/routers/users.py:55]: Can you move it to `config.py` then

File: frontend/src/contexts/UserContext.tsx (hunk 1 above)
Comment (taehyoungjo) [reply frontend/src/contexts/UserContext.tsx:112]: It doesn't have to be

**Pull request merged**